numpy/core/src/multiarray/mapping.c

```diff
@@ -2134,6 +2134,7 @@ array_assign_subscript(PyArrayObject *self, PyObject *ind, PyObject *op)
 
     /* Clean up temporary variables and indices */
   fail:
+    PyArray_DiscardWritebackIfCopy(view);
```

Comment:
I'm assuming `view` can be `NULL` here from the `XDECREF` below. Is this save if `view` is `NULL`?

Reply:
``PyArray_DiscardWritebackIfCopy(arr)`` does check if ``arr`` is ``NULL``